### PR TITLE
Support broadcast signal for signal intermediate throw events

### LINKED ISSUE #11918: [EPIC] Broadcast signal event using throw signal event
## Introduction

 This is stage 3 of the umbrella epic:
 
 * [[EPIC] Support signal events #10777](https://github.com/camunda/zeebe/issues/10777)

 ### Stage 3: Broadcast signal event using throw signal event

In the third stage, we'll increase support to all BPMN signal symbols. Specifically, it adds support for the Signal Intermediate Throw Event, and the Signal End Event.

Model highlighting that all the signal events will be supported at this stage, including the signal throw events

When a process instance arrives at a signal throw event, we'll broadcast a signal the same way as the gateway can: write a `Signal:Broadcast` command to the current partition. The implementation of stage 1 and 2 will then distribute the command to other partitions, and all signal (start event) subscriptions will be triggered, without additional implementation efforts needed.

### Tasks

- [x] #11919
	- When Signal Intermediate Throw Event activates:
		- Apply input mappings
		- Transition to activated
		- Write `Signal:Broadcast` command
		- Apply output mappings
		- Transition to complete the element
- [x] #11920
	- Introduce a new `SignalEndEventBehavior`
	- When Signal End Event activates:
		- Apply input mappings
		- Transition to activated
		- Write `Signal:Broadcast` command
		- Apply output mappings
		- Transition to complete the element

 

### Discussions

<details>
<summary>Signal name as an expression is a stretch goal</summary>
<p>

By default, the signal name is a static value. This adds value to most users. Expression support for the signal name can be added later. As in other places, this would simply require the value to be prefixed with the expression marker `=`. See stage 4.

</p>
</details>

<details>
<summary>Broadcasting signals with variables payload is a stretch goal</summary>
<p>

The `Signal` record value will already be made ready to receive variables, but we'll leave them out of the gRPC message. This way, it will not yet be possible to pass variables along with the signal, but we'll be able to extend this in the future. Variables are useful but do not have to be part of the first iteration of this feature. See stage 4.

</p>
</details>

## Follow-up

Once the above tasks are completed, we need to
- [ ] Document signal end events
- [ ]  create an issue for Modeler to support signal end events
- [ ]  create an issue for Operate to support signal end events
- [ ]  create an issue for Optimize to support signal end events
- [ ]  inform the product-hub about the completion

## Links

- Product Hub (internal): https://github.com/camunda/product-hub/issues/657

### LINKED ISSUE #11919: Support Broadcast signal for Signal Intermediate Throw Events
- ## Adjust `IntermediateThrowEventProcessor` to broadcast a signal on activation
	- When Signal Intermediate Throw Event activates:
		- Apply input mappings
		- Transition to activated
		- Write `Signal:Broadcast` command
		- Apply output mappings
		- Transition to complete the element

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Support broadcast signal for signal intermediate throw events, also need unblock the signal intermediate throw event of https://github.com/camunda/zeebe/pull/12117 and delete the `UnsupportedSignalThrowEventValidator` class.

- When Signal Intermediate Throw Event activates:
  - Apply input mappings
  - Transition to activated
  - Write `Signal:Broadcast` command
  - Apply output mappings
  - Transition to complete the element

## Related issues

closes #11919
closes #11918

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [x] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by skayliu** [comment]
Hey, @korthout. Please take a moment to check this out, When you have time. 🙇

File: engine/src/test/java/io/camunda/zeebe/engine/processing/signal/SignalThrowEventTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import io.camunda.zeebe.protocol.record.value.BpmnEventType;
+import io.camunda.zeebe.protocol.record.value.VariableRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class SignalThrowEventTest {
```
Comment (korthout) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/signal/SignalThrowEventTest.java:30]: 🔧 Let's rename this to `SignalIntermediateThrowEventTest` to clarify that this only tests the Intermediate Throw Events, not also the End Events.

**Review by korthout — CHANGES_REQUESTED** [review]
As I've said before 👏 Well done @skayliu 

👍 This looks great!

❌ I only have some minor remarks. Please have a look and make the requested change. Then we should be able to merge this soon after.

**Review by korthout — APPROVED** [review]
Thanks @skayliu 🥇 

LGTM 👍 

bors merge

**Pull request merged**